pkg/report/sarif.go

```diff
@@ -0,0 +1,224 @@
+					vuln.VulnerabilityID, vuln.Severity, vuln.PkgName, vuln.FixedVersion, vuln.VulnerabilityID, vuln.PrimaryURL, vuln.Description),
+				helpMarkdown: fmt.Sprintf("**Vulnerability %v**\n| Severity | Package | Fixed Version | Link |\n| --- | --- | --- | --- |\n|%v|%v|%v|[%v](%v)|\n\n%v",
+					vuln.VulnerabilityID, vuln.Severity, vuln.PkgName, vuln.FixedVersion, vuln.VulnerabilityID, vuln.PrimaryURL, vuln.Description),
+				message: fmt.Sprintf("Package: %v\nInstalled Version: %v\nVulnerability %v\nSeverity: %v\nFixed Version: %v\nLink: [%v](%v)",
+					vuln.PkgName, vuln.InstalledVersion, vuln.VulnerabilityID, vuln.Severity, vuln.FixedVersion, vuln.VulnerabilityID, vuln.PrimaryURL),
+			})
+		}
+		for _, misconf := range res.Misconfigurations {
+			sw.addSarifResult(&sarifData{
+				vulnerabilityId:  misconf.ID,
+				severity:         misconf.Severity,
+				url:              misconf.PrimaryURL,
+				resourceType:     res.Type,
+				artifactLocation: toPathUri(res.Target),
+				resultIndex:      getRuleIndex(misconf.ID, ruleIndexes),
+				fullDescription:  html.EscapeString(misconf.Description),
+				helpText: fmt.Sprintf("Misconfiguration %v\nType: %s\nSeverity: %v\nCheck: %v\nMessage: %v\nLink: [%v](%v)\n%s",
+					misconf.ID, misconf.Type, misconf.Severity, misconf.Title, misconf.Message, misconf.ID, misconf.PrimaryURL, misconf.Description),
+				helpMarkdown: fmt.Sprintf("**Misconfiguration %v**\n| Type | Severity | Check | Message | Link |\n| --- | --- | --- | --- | --- |\n|%v|%v|%v|%s|[%v](%v)|\n\n%v",
+					misconf.ID, misconf.Type, misconf.Severity, misconf.Title, misconf.Message, misconf.ID, misconf.PrimaryURL, misconf.Description),
+				message: fmt.Sprintf("Artifact: %v\nType: %v\nVulnerability %v\nSeverity: %v\nMessage: %v\nLink: [%v](%v)",
+					res.Target, res.Type, misconf.ID, misconf.Severity, misconf.Message, misconf.ID, misconf.PrimaryURL),
+			})
+		}
+	}
+	sw.run.ColumnKind = columnKind
+	sw.run.OriginalUriBaseIDs = map[string]*sarif.ArtifactLocation{
+		"ROOTPATH": {URI: &rootPath},
+	}
+	sarifReport.AddRun(sw.run)
+	return sarifReport.PrettyWrite(sw.Output)
+}
+
+func toSarifRuleName(vulnerabilityType string) string {
+	switch vulnerabilityType {
+	case os.RedHat, os.Debian, os.Ubuntu, os.CentOS, os.Rocky, os.Alma, os.Fedora,
+		os.Amazon, os.Oracle, os.OpenSUSE, os.OpenSUSELeap, os.OpenSUSETumbleweed, os.SLES,
+		os.Photon, os.Alpine, os.Windows:
+		return sarifOsPackageVulnerability
+
+	case types.Bundler, types.GemSpec, types.Cargo, types.Composer, types.Npm, types.NuGet, types.Pip,
+		types.Pipenv, types.Poetry, types.PythonPkg, types.NodePkg, types.Yarn, types.Jar, types.Pom,
+		types.GoBinary, types.GoMod, types.JavaScript:
+		return sarifLanguageSpecificVulnerability
+
+	case types.YAML, types.JSON, types.TOML, types.Dockerfile, types.HCL, types.Terraform,
+		types.Kubernetes, types.CloudFormation, types.Ansible:
+		return sarifConfigFiles
+
+	default:
+		return sarifOtherVulnerability
+	}
+}
```

Comment:
I meant this switch can be simplified

```suggestion
func toSarifRuleName(class string) string {
	switch class {
	case ClassOSPkg:
		return sarifOsPackageVulnerability
	case ClassLangPkg:
		return sarifLanguageSpecificVulnerability
	case ClassConfig:
		return sarifConfigFiles
	default:
		return sarifOtherVulnerability
	}
}
```

Reply:
oh, ok. changed